crates/uv-fs/src/lib.rs

```diff
@@ -40,7 +58,22 @@ pub fn read_to_string(path: impl AsRef<Path>) -> std::io::Result<String> {
         std::io::stdin().read_to_string(&mut buf)?;
         Ok(buf)
     } else {
-        fs::read_to_string(path)
+        if path.starts_with("http://") | path.starts_with("https://") {
+            let path_utf8 = path.to_str().ok_or_else(|| {
+                std::io::Error::new(
+                    std::io::ErrorKind::InvalidInput,
+                    format!("Path '{:?}' is not valid unicode.", &path),
+                )
+            })?;
+            Ok(reqwest::get(path_utf8)
+                .await
+                .map_err(|err| std::io::Error::new(std::io::ErrorKind::Other, err))?
+                .text()
+                .await
+                .map_err(|err| std::io::Error::new(std::io::ErrorKind::Other, err))?)
+        } else {
+            Ok(tokio::read_to_string(path).await?)
+        }
```

Comment:
I think I would prefer to add dedicated methods for this in `requirements-txt`, for two reasons:

1. I don't think we want to allow users to provide `pyproject.toml` files as URLs for now. I'd just prefer to limit it to requirements files since no tool supports remote `pyproject.toml` (and, in practice, we're actually supposed to _build the project_ when we're given a `pyproject.toml`, so this wouldn't work correctly in all cases).
2. Adding `reqwest` as a dep to `uv-fs` feels heavy since this crate is supposed to be lightweight.

Reply:
implemented in [1ef5aac](https://github.com/astral-sh/uv/pull/2081/commits/1ef5aacfb0064022b9273a0248448d232bbbb946)